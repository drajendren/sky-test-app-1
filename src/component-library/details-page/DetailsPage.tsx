import React from "react";
import { useLocation } from "react-router-dom";
import { MovieDetails, PersonDetails, TvDetails } from "..";
import { Movie, Person, TvShow } from "../../interfaces";

export const DetailsPage = () => {
  const state = useLocation().state as any;
  const props = state.props;
  return (
    <div>
      {props.media_type === "tv" ? (
        <TvDetails details={props as TvShow} />
      ) : props.media_type === "movie" ? (
        <MovieDetails details={props as Movie} />
      ) : props.media_type === "person" ? (
        <PersonDetails details={props as Person} />
      ) : null}
    </div>
  );
};

export default DetailsPage;
