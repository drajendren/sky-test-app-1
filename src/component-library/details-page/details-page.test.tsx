import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import DetailsPage from "./DetailsPage";

const detailsPageData = {props: {
    backdrop_path: "/iIQKA17YHJUmGYRpXmP0KXaxq99.jpg",
    first_air_date: "1996-08-26",
    genre_ids: [
      10759,
      16
    ],
    id: 1168,
    media_type: "tv",
    name: "The Real Adventures of Jonny Quest",
    origin_country: [
      "US"
    ],
    original_language: "en",
    original_name: "The Real Adventures of Jonny Quest",
    overview: "The Real Adventures of Jonny Quest is an American animated action-adventure television series.",
    popularity: 23.65,
    poster_path: "/vx8KdVNOpHVBR0x30shg5kcGnd7.jpg",
    vote_average: 7.1,
    vote_count: 26
  }};
  
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
      pathname: "localhost:3000/123",
      state: detailsPageData
    })
  }));
describe("<DetailsPage />", () => {
    
  test("it should display correct details page from state", () => {
    const { baseElement } = render(<DetailsPage />);
    expect(baseElement).toMatchSnapshot();
  });
});
