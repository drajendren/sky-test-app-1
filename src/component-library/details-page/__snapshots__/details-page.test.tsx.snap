// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsPage /> it should display correct details page from state 1`] = `
<body>
  <div>
    <div>
      <h2>
        Tv show Details
      </h2>
      <h3>
        Name
      </h3>
      <div>
        The Real Adventures of Jonny Quest
      </div>
      <h3>
        Genre
      </h3>
      <div>
        Animation
      </div>
      <h3>
        Overview
      </h3>
      <div
        class="overview"
      >
         
        The Real Adventures of Jonny Quest is an American animated action-adventure television series.
      </div>
      <h3>
        First Aired
      </h3>
      <div>
        1996-08-26
      </div>
      <img
        alt=""
        height="200px"
        src="https://image.tmdb.org/t/p/w500/vx8KdVNOpHVBR0x30shg5kcGnd7.jpg"
      />
    </div>
  </div>
</body>
`;
